src/java/us/kbase/templates/baseclient.py

```diff
@@ -236,14 +239,22 @@ def run_job(self, service_method, args, service_ver=None, context=None):
         mod, _ = service_method.split('.')
         job_id = self._submit_job(service_method, args, service_ver, context)
         async_job_check_time = self.async_job_check_time
-        while True:
+        check_job_failures = 0
+        while check_job_failures < _CHECK_JOB_RETRYS:
             time.sleep(async_job_check_time)
             async_job_check_time = (async_job_check_time *
                                     self.async_job_check_time_scale_percent /
                                     100.0)
             if async_job_check_time > self.async_job_check_max_time:
                 async_job_check_time = self.async_job_check_max_time
-            job_state = self._check_job(mod, job_id)
+
+            try:
+                job_state = self._check_job(mod, job_id)
+            except ConnectionError:
+                _traceback.print_exc()
+                check_job_failures += 1
+                continue
+
```

Comment:
Maybe want to log that checkjob failed if check_job_failures == _CHECK_JOB_RETRYS

Reply:
Actually need to raise and exception at that point. Thx!